src/Microsoft.Extensions.Options/IOptions.cs

```diff
@@ -7,11 +7,20 @@ namespace Microsoft.Extensions.Options
     /// Used to retreive configured TOptions instances.
     /// </summary>
     /// <typeparam name="TOptions">The type of options being requested.</typeparam>
-    public interface IOptions<out TOptions> where TOptions : class, new()
+    public interface IOptions<TOptions> where TOptions : class, new()
```

Comment:
>  still think maybe we should consider just adding Get(name) to IOptions, but not break IConfigureOptions as even though we are changing the IOptions interface), there's not many implementations of IOptions today, while implementing your own IConfigureOptions was a major extensibility point which we should avoid breaking

Ah, so you are actually proposing keeping the `Value` property and adding the method alongside it, correct? Otherwise we would break all consumers of `IOptions`. I don't like it being a property if it will potentially return a different instance sometimes when invoked, but it may be just me.